### PR TITLE
testting diffrence function

### PR DESCRIPTION
**Pull request by Ibrahim86Ali**
## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [x] The function's name is clear and helpful
  - [x] The parameter names are clear and helpful
  - [x] Default parameters are used if possible
- **JSDoc Comment**
  - [x] The docstring is well-formatted
  - [x] There are no spelling mistakes
  - [x] The description of the function's behavior is clear
  - [ ] All parameters are documented with types
  - [ ] Documented parameter names match the function's parameter names
  - [x] Parameters have a good description
  - [ ] If default parameters are used, they are the correct type
  - [ ] The return value is documented with a type
  - [ ] The return value has a clear description
- **README**
  - [x] All the sections in the solution README are complete
  - [x] Descriptions in the README are thorough and clear
  - [x] There is at least 2 use-cases
- **Unit Tests**
  - [x] Unit tests are present
  - [ ] There are at least 20 test cases
  - [ ] All `describe`s and `it`s have clear and helpful names
  - [ ] All test names actually describe what is being tested
  - [ ] Tests are grouped logically using `describe`
  - [x] There is only one `expect` inside every `it`
  - [x] There is no logic (loops, conditionals, ...) in the test cases
  - [x] Side-effects are tested, if necessary
  - [x] All tests pass in the CI check
  - [x] The tests have 100% coverage of the solution
  - [ ] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [ ] Variables have names that help understand the strategy
  - [ ] Spacing and formatting help to understand the strategy
  - [ ] The code is written as simply and clearly as possible
  - [ ] Use short, clear comments to describe the strategy
- **README**
  - [x] There is an explanation of the solution's strategy in the README
  - [x] The explanation is clear and understandable
  - [x] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [x] No formatting errors appear in the linting check
- **Linting**
  - [x] There should be no linting errors or warnings in the CI check
  - [x] Spelling should be correct in all code, comments and docstrings
- **The solution file**
  - [ ] Has a JSDoc comment
  - [x] Has an exported function
  - [x] Does no other code or comments outside of the function body (you can use
        `sandbox.js` for saving extra ideas)
- **The function body _Should_ ...**
  - [ ] Be well and consistently formatted
  - [x] Have clear and helpful variable names
  - [x] Declare variables in the smallest scope possible
  - [x] Be a simple and readable implementation
- **The function body should _Not_ ...**
  - [x] Have extra comments that are not about the solution's strategy
  - [x] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [ ] Have any `console` calls (`.log`, `.assert`, ...)
  - [x] Have any `debugger` statements

### CODE REVIEW

File: .template-solution/difference.spec.js [hunk 1]
```diff
@@ -9,4 +9,6 @@ it('should not repeat return values  ', () => {
   expect(difference([1, 2, 1], [2, 3])).toEqual([1]);
 });
 
+;
```
Comment (HerbertHomolka1) [inline .template-solution/difference.spec.js:12]: remove the semi-colon

File: .template-solution/difference.spec.js [hunk 2]
```diff
@@ -9,4 +9,6 @@ it('should not repeat return values  ', () => {
   expect(difference([1, 2, 1], [2, 3])).toEqual([1]);
```
Comment (HerbertHomolka1) [inline .template-solution/difference.spec.js:9]: add more tests.
test for repeated numbers within either array

File: .template-solution/difference.js [hunk 3]
```diff
@@ -16,5 +16,7 @@ export const difference = (array = [], filterValues = []) => {
 
 console.log(difference([1,3,3],[1]))
```
Comment (HerbertHomolka1) [inline .template-solution/difference.js:17]: remove the console log

File: .template-solution/difference.js [hunk 4]
```diff
@@ -16,5 +16,7 @@ export const difference = (array = [], filterValues = []) => {
...
 
+;
```
Comment (HerbertHomolka1) [inline .template-solution/difference.js:19]: remove the semi-colon

File: .template-solution/difference.js [hunk 5]
```diff
@@ -16,5 +16,7 @@ export const difference = (array = [], filterValues = []) => {
```
Comment (HerbertHomolka1) [inline .template-solution/difference.js:16]: run prettier. You have too much whitespace.

**Review by HerbertHomolka1 — COMMENTED** [review]
overall nice. code works. Run prettier and remove the console log/ unnecessary semi-colons
